test/jasmine/tests/drawing_test.js

```diff
@@ -348,6 +345,54 @@ describe('Drawing', function() {
             expect(g.attr('transform')).toEqual('translate(8,9) scale(4,5) translate(-8,-9) translate(1, 2)');
         });
     });
+
+    describe('bBox', function() {
+        afterEach(destroyGraphDiv);
+
+        it('should update bounding box dimension on window scroll', function(done) {
+            var gd = createGraphDiv();
+
+            // allow page to scroll
+            gd.style.position = 'static';
+
+            Plotly.plot(gd, [{
+                y: [1, 2, 1]
+            }], {
+                annotations: [{
+                    text: 'hello'
+                }],
+                height: window.innerHeight * 2,
+                width: 500
+            })
+            .then(function() {
+                var node = d3.select('text.annotation').node();
+                expect(Drawing.bBox(node)).toEqual({
+                    height: 14,
+                    width: 27.671875,
+                    left: -13.671875,
+                    top: -11,
+                    right: 14,
+                    bottom: 3
+                });
+
+                window.scroll(0, 200);
+                return Plotly.relayout(gd, 'annotations[0].text', 'HELLO');
+            })
+            .then(function() {
+                var node = d3.select('text.annotation').node();
+                expect(Drawing.bBox(node)).toEqual({
+                    height: 14,
+                    width: 41.015625,
+                    left: -20.671875,
+                    top: -11,
+                    right: 20.34375,
+                    bottom: 3
```

Comment:
Commit https://github.com/plotly/plotly.js/pull/1585/commits/29abb5d8ec3dea92cc71f438fffa6785463d1fb1 made the `bottom` and `top` keys here incorrectly change - as the bbox computation were based on a outdated cached tester element.